ml-agents/mlagents/trainers/subprocess_env_manager.py

```diff
@@ -137,9 +153,21 @@ def _generate_all_results() -> AllStepResult:
         return all_step_result
 
     try:
-        env = env_factory(
-            worker_id, [env_parameters, engine_configuration_channel, stats_channel]
-        )
+        side_channels = [env_parameters, engine_configuration_channel, stats_channel]
+        if training_analytics_channel is not None:
```

Comment:
I would put this line after line 161 (after we check if the environment can support training_analytics

Reply:
Agreed it's a bit convoluted. But we don't currently have a way to add a SideChannel after the BaseEnv has been created, and we can't get the capabilities until after we've created the environment. So this always adds the sidechannel (for worker==0) but then doesn't use it if the capabilities don't support it.

Do you think it's worth adding an interface to BaseEnv to add SideChannels after creation? Note that modifying the `side_channels` list after creation won't have any effect, because the list is turned into an ID->SideChannel mapping in https://github.com/Unity-Technologies/ml-agents/blob/728d4927a820e43b398f0738434fc070dec33e97/ml-agents-envs/mlagents_envs/side_channel/side_channel_manager.py#L65